src/m365/flow/commands/owner/owner-add.spec.ts

```diff
@@ -0,0 +1,193 @@
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      aadGroup.getGroupByDisplayName,
+      aadUser.getUserIdByUpn,
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      telemetry.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.OWNER_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if name is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { environmentName: validEnvironmentName, name: 'invalid', userId: validUserId, roleName: validRoleName } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if userId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { environmentName: validEnvironmentName, name: validName, userId: 'invalid', roleName: validRoleName } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if groupId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { environmentName: validEnvironmentName, name: validName, groupId: 'invalid', roleName: validRoleName } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if username is not a valid user principal name', async () => {
+    const actual = await command.validate({ options: { environmentName: validEnvironmentName, name: validName, userName: 'invalid', roleName: validRoleName } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if roleName is not a valid user principal name', async () => {
+    const actual = await command.validate({ options: { environmentName: validEnvironmentName, name: validName, userName: validUserName, roleName: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the username passed', async () => {
+    const actual = await command.validate({ options: { environmentName: validEnvironmentName, name: validName, userName: validUserName, roleName: validRoleName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('adds owner to the flow with userId', async () => {
+    sinon.stub(request, 'post').callsFake(async opts => {
+      if (opts.url === `https://management.azure.com/providers/Microsoft.ProcessSimple/environments/${formatting.encodeQueryParameter(validEnvironmentName)}/flows/${formatting.encodeQueryParameter(validName)}/modifyPermissions?api-version=2016-11-01`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.doesNotReject(command.action(logger, { options: { verbose: true, environmentName: validEnvironmentName, name: validName, userId: validUserId, roleName: 'CanView' } }));
```

Comment:
If I'm not mistaken, `assert.doesNotReject` is not really needed here. If the command fails, an error is thrown, because the error is not catched in the test, the whole test will fail.
Same for other tests.